src/components/vehicleReference/index.tsx

```diff
@@ -1,33 +1,33 @@
 import React, { ComponentProps, FC, ReactNode } from 'react';
 
-import { Box, chakra, useMultiStyleConfig } from '@chakra-ui/react';
+import { chakra, useSlotRecipe } from '@chakra-ui/react';
 
-import Stack from '../stack';
+import { AspectRatio, Box, Grid, MissingImage, Stack } from 'src';
 
-import MissingImage from '../missingImage';
-import Grid from '../grid';
-import AspectRatio from '../aspectRatio';
-
-interface Props {
+export type VehicleReferenceProps = {
   image?: ReactNode;
   vehicleTitle: string;
   price?: string | null;
   sellerName?: string | null;
   sellerAddress?: string | null;
   callToAction?: ReactNode;
   templateColumns?: ComponentProps<typeof Grid>['templateColumns'];
-}
+};
 
-const VehicleReference: FC<Props> = ({
-  image,
-  vehicleTitle,
-  price,
-  sellerName,
-  sellerAddress,
-  callToAction,
-  templateColumns = { base: 'auto 1fr', md: '1fr' },
-}) => {
-  const styles = useMultiStyleConfig(`VehicleReference`);
+export const VehicleReference: FC<VehicleReferenceProps> = (props) => {
+  const recipe = useSlotRecipe({ key: 'VehicleReference' });
+  const [recipeProps, componentProps] = recipe.splitVariantProps(props);
+  const styles = recipe(recipeProps);
+
+  const {
+    image,
+    vehicleTitle,
+    price,
+    sellerName,
+    sellerAddress,
+    callToAction,
+    templateColumns = { base: 'auto 1fr', md: '1fr' },
```

Comment:
Word of warning - if you split and assign like this then storybook won't see the values as default props. It's safe to destructure non-theme props in the argument definition (Line 17)